### PR TITLE
Use Vuex for job theme (?), react on changes in views & layouts, and …

### PR DESCRIPTION
**Pull request by kinow**
…pass the theme as a prop in the component

@oliver-sanders not the best way to achieve this I think, just started adding changes until I got the theme set in Vuex, and other views/layout reacting to this value.

Then set the theme as a prop in the `Job` component.

I have been trying to keep Vuex away from components, and using it only in views. This forces the components to have as little state as possible, and zero Vuex, which IMO has made maintaining these components easier.

But even if we wanted to use Vuex directly in the `Job` component, I don't know if that would work, as `Job` is a functional component (only props, no data, no computed). The functional marker enforces the immutability on the component, and also enforces the Vue design of props-down (and events-up).

Feel free to use any parts of this PR if you'd like :+1: it was just to show how I would approach it more or less

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Thanks for taking a look at this, I like the VueX approach.

I'm not keen on setting the theme locally and expecting every view that uses the Job component to read the theme and set it accordingly. (e.g. when we enable users to add their own views they would have to explicitly support job colour themes).

Could we get the Job component to read the theme from VueX (keeping the prop to allow it to be overridden for the UserProfile page)? Or does that break the responsive model.

**Comment by kinow** [comment]
>Could we get the Job component to read the theme from VueX (keeping the prop to allow it to be overridden for the UserProfile page)? Or does that break the responsive model.

I think we would need to investigate a bit more. A good starting point is this docs: https://vuejs.org/v2/guide/render-function.html#Functional-Components

>In cases like this, we can mark components as functional, which means that they’re stateless (no reactive data) and instanceless (no this context). A functional component looks like this:

From what I remember `Job` should be rendered faster for being functional, but on the other hand we must pass everything via props as it doesn't have `data`, `computed`, nor `this`. 

This example suggests an approach that would work for us if we were not using a functional template: https://forum.vuejs.org/t/how-can-i-access-vuex-getters-or-state-in-functional-component/16806/2

IOW, we would have to delete the `<template` from `Job`, and re-write the exact same components with the JSX syntax for functional components (similar to React). That way I think we could access the `store` as in that example. Or we can try to access the `store` from the template, though I think I may have tried that some time ago and I think accessing imports directly in templates does not work… instead you must first bind them to a `data` element?

**Comment by oliver-sanders** [comment]
Ok, and I agree with you that keeping logic out of the components is fundamentally a good thing.

In which case, how about a computed property on the application for setting the class list? That keeps it simple and CSS, no DOM re-rendering necessary (after the one class change in #app).

File: src/layouts/Default.vue [hunk 1]
```diff
@@ -70,7 +72,10 @@ export default {
   },
 
   computed: {
-    ...mapState(['offline'])
+    ...mapState(['offline']),
+    ...mapState({
+      jobTheme: state => `job_theme--${state.app.jobTheme}`
```
Comment (kinow) [inline src/layouts/Default.vue:77]: This `computed` property `jobTheme` contains the app job theme @oliver-sanders . Doesn't need to be necessarily here, but just to show where I had set it (though I just realized I used some cryptic syntax, not sure why... quite sure I can write just `mapState('app', ['jobTheme'])` instead; was probably running low on :coffee: )

**Comment by kinow** [comment]
>In which case, how about a computed property on the application for setting the class list? That keeps it simple and CSS, no DOM re-rendering necessary (after the one class change in #app).

Sounds good to me. I did something similar (if I understand correctly, near midnight here) in the layout I think (or in some view, will mark it in a comment).

**Comment by oliver-sanders** [comment]
Thanks for your help, I'm trying to get back into Vue after a long leave of absence and am still a bit shaky on things.

**Pull request merged**